packages/editor/src/plugins/drag-drop-bg/components/answer-zone/AnswerZone.tsx

```diff
@@ -0,0 +1,151 @@
+  getAnswerZoneText: (text: string) => string
+  onChangeDimensions: (
+    id: string,
+    dimensions: { width: number; height: number }
+  ) => void
+}
+
+export const AnswerZone = forwardRef<HTMLDivElement, AnswerZoneProps>(
+  (
+    {
+      answerZone,
+      isDraggingEnabled,
+      onClickSettingsButton,
+      onClickPlusButton,
+      getAnswerZoneImageSrc,
+      onChangeDimensions,
+      getAnswerZoneText,
+    },
+    ref
+  ) => {
+    const [isResizing, setIsResizing] = useState(false)
+    const [dimensions, setDimensions] = useState({ width: 200, height: 70 })
+
+    const [{ isDragging }, drag, dragPreview] = useDrag(
+      () => ({
+        type: 'all',
+        item: answerZone,
+        canDrag: isDraggingEnabled && !isResizing,
+        collect: (monitor) => ({
+          isDragging: monitor.isDragging(),
+        }),
+      }),
+      [answerZone, isDraggingEnabled, isResizing]
+    )
+
+    // TODO: Check exactly what this does
+    useImperativeHandle(ref, () => ({
+      getBoundingClientRect: () => dimensions,
+    }))
+
+    const handleResize: ResizableBoxProps['onResize'] = (_, { size }) => {
+      setIsResizing(true)
+      setDimensions(size)
+      onChangeDimensions(answerZone.id.get(), size)
+    }
+
+    const handleResizeStop: ResizableBoxProps['onResizeStop'] = () => {
+      setIsResizing(false)
+    }
+
+    const renderButtons = () => (
+      <div
+        style={{
+          ...buttonsContainerStyle,
+          height: dimensions.height,
+          width: dimensions.width,
+        }}
+      >
+        <button
+          style={settingsButtonStyle}
+          onClick={() => onClickSettingsButton?.(answerZone.id.get())}
+        >
+          <FaIcon icon={faCog} />
+        </button>
+        <button
+          style={plusButtonStyle}
+          onClick={() => onClickPlusButton?.(answerZone.id.get())}
+        >
+          <FaIcon icon={faPlus} />
+        </button>
+      </div>
+    )
+
+    const left = answerZone.position.left.get()
+    const top = answerZone.position.top.get()
+    const answerImageUrl = getAnswerZoneImageSrc(answerZone.answer.image.get())
+    const answerText = getAnswerZoneText(answerZone.answer.text.get())
+
+    return (
+      <div
+        ref={dragPreview}
+        style={{
+          ...answerZoneStyle,
+          left,
+          top,
+          width: dimensions.width,
+          height: dimensions.height,
+        }}
+        data-testid="box"
+      >
+        <ResizableBox
+          width={dimensions.width}
+          height={dimensions.height}
+          minConstraints={[100, 50]}
+          maxConstraints={[500, 300]}
+          onResize={handleResize}
+          onResizeStop={handleResizeStop}
+          resizeHandles={['nw', 'ne', 'sw', 'se']}
+          style={{ width: '100%', height: '100%' }}
+        >
+          <div
+            ref={drag}
+            style={{ width: '100%', height: '100%', position: 'relative' }}
```

Comment:
please use tailwind where feasible. you can get an introduction from any of the frontend people.

Reply:
Done :heavy_check_mark: